src/main/kotlin/OSSLLMCompletionProvider.kt

```diff
@@ -1,17 +1,39 @@
 package com.github.bzz.intellij
 
+import com.github.bzz.intellij.requests.Requester
 import com.intellij.codeInsight.inline.completion.InlineCompletionElement
-import com.intellij.codeInsight.inline.completion.InlineCompletionProvider;
+import com.intellij.codeInsight.inline.completion.InlineCompletionProvider
 import com.intellij.codeInsight.inline.completion.InlineCompletionRequest
+import com.intellij.openapi.application.runReadAction
+import com.intellij.openapi.diagnostic.thisLogger
 import com.intellij.openapi.editor.event.DocumentEvent
+import java.io.IOException
 
 class OSSLLMCompletionProvider : InlineCompletionProvider {
 
     override suspend fun getProposals(request: InlineCompletionRequest): List<InlineCompletionElement> {
-        return listOf(InlineCompletionElement("public static void"),InlineCompletionElement("this is a test \uD83E\uDD16"),)
+        try {
+            val caretPosition = runReadAction {
+                request.editor.caretModel.primaryCaret.selectionStart
+            }
+            val proposal = Requester.getModelSuggestions(
+                request.document.text.substring(0, caretPosition)
+            ).text ?: return emptyList()
+            return listOf(InlineCompletionElement(proposal))
+        } catch(exception: RuntimeException) {
+            thisLogger().warn("Message: ${exception.message}")
```

Comment:
Nit pick: by looking at logs generated with this code "Message: HTTP Request failed. Failure status code 404" - how would you be able to tell the source of the error?

E.g. it could be nice to mention that it is a request to LLM for code completion that has failed.